frontend/components/ifixit-image/ifixit-image.tsx

```diff
@@ -66,12 +66,28 @@ function isStrapiImage(src: string) {
    );
 }
 
-function getImageLoader(sizeMap: SizeMap, defaultSize: string): ImageLoader {
+function getImageLoader(
+   sizeMap: SizeMap,
+   defaultSize: string,
+   baseWidth?: number | string
+): ImageLoader {
    return ({ src, width }: ImageLoaderProps) => {
       const baseSrc = src.replace(/\.[^/.]+$/, '');
-      const sizeName = getImageSize(width, sizeMap, defaultSize);
+      let realWidth = width;
+      if (typeof baseWidth === 'string') {
+         realWidth = parseInt(baseWidth);
+      } else if (baseWidth) {
+         realWidth = baseWidth;
+      }
+      // If next wants to see the 2x image here then this accounts for that in our calculations of resolution
+      // see next's image.js getWidths() for more details
+      if (width / realWidth > 2) {
+         realWidth *= 2;
+      }
+
+      const sizeName = getImageSize(realWidth, sizeMap, defaultSize);
       // We don't use the ?width param server-side, but it gets rid of a nextjs warning
-      return baseSrc.concat('.', sizeName, `?width=${width}`);
+      return baseSrc.concat('.', sizeName, `?width=${realWidth}`);
```

Comment:
I think it may be a good idea to:

1. Cast to a URL (new URL(string))
2. Add a few jest tests around this function

The issue seems to indicate we were generating non-urls:

`GUID.size400?width=384`

Given the last line `baseSrc.concat('.',...` I'd bet I could break this just by passing a src with a query param:

`const src = "www.ifixit.com?withSomeParam=1"` would like end up close to `www.ifixit.com?withSomeParam=1?width=100` which is not a valid url

Reply:
This issue isn't that we are generating non-urls, its that our databases of pictures aren't complete. Next js basically expects us to have our pictures available in various sizes, and our pictures are not standardized to having any specific sizes available as far as I'm aware. 

> Given the last line baseSrc.concat('.',... I'd bet I could break this just by passing a src with a query param:

I think the picture sources are coming out of algolia, not provided by the user